integration-tests/digitalocean/src/test/java/org/apache/camel/quarkus/component/digitalocean/it/DigitaloceanDropletTest.java

```diff
@@ -0,0 +1,451 @@
+                .get("/digitalocean/droplet/ipv6/" + dropletId)
+                .then()
+                .body("resourceId", equalTo(dropletId));
+    }
+
+    /**
+     * Test the result of actions performed
+     *
+     */
+    private void testResultActions(Integer dropletId) {
+        // getting the droplet actions
+        List<Map<String, Object>> actions = RestAssured.given().contentType(ContentType.JSON)
+                .get("/digitalocean/droplet/actions/" + dropletId)
+                .then().extract().body().as(List.class);
+        List<String> types = Arrays.asList("ENABLE_BACKUPS", "DISABLE_BACKUPS", "SNAPSHOT", "POWER_ON", "POWER_OFF", "REBOOT",
+                "ENABLE_IPV6");
+        assertActions(actions, types);
+
+        // getting the snapshot action id
+        Integer snapshotActionId = getSnapshotActionId(actions);
+
+        // getting one action
+        given()
+                .get("/digitalocean/action/" + snapshotActionId)
+                .then()
+                .body("id", equalTo(snapshotActionId));
+
+        // getting all actions on the account
+        actions = RestAssured.given().contentType(ContentType.JSON)
+                .get("/digitalocean/actions/")
+                .then().extract().body().as(List.class);
+        types = Arrays.asList("ENABLE_BACKUPS", "DISABLE_BACKUPS", "SNAPSHOT", "POWER_ON", "POWER_OFF", "REBOOT");
+        assertActions(actions, types);
+    }
+
+    /**
+     * Gets the snapshots and waits until the snapshot is created in Digitalocean.
+     *
+     */
+    private void waitForSnapshot(Integer dropletId) {
+        await().atMost(timeout, timeoutUnit).until(() -> {
+            String path = "/digitalocean/droplet/snapshots/" + dropletId;
+            List<Map<String, Object>> result = given().when().get(path).then().extract().as(List.class);
+            // Look for the snapshot
+            Optional<Map<String, Object>> optional = result.stream()
+                    .filter(s -> "snapshot1".equals(s.get("name")))
+                    .findAny();
+            return optional.isPresent();
+        });
+    }
+
+    /**
+     * Gets the droplet and waits until the droplet is Active in Digitalocean.
+     *
+     */
+    private void waitActiveDroplet(Integer dropletId) {
+        await().atMost(timeout, timeoutUnit).until(() -> {
+            String path = "/digitalocean/droplet/" + dropletId;
+            Map<String, Object> droplet = given()
+                    .contentType(ContentType.JSON).get(path).then().extract().as(Map.class);
+            return droplet != null && dropletId.equals(droplet.get("id")) && "ACTIVE".equals(droplet.get("status"));
+        });
+    }
+
+    /**
+     * Assert all the actions
+     *
+     */
+    private void assertActions(List<Map<String, Object>> actions, List<String> types) {
+        // verify there are actions
+        assertNotNull(actions);
+        // verify there are at least the 7 created actions in the test
```

Comment:
So, I guess it's not always 7 in the context of this function.

Reply:
Yes there can be some other actions done by Digitalocean for instance (internal process); So I was just testing the actions that we know we've done in this test suit